src/util/replace_symbol.cpp

```diff
@@ -89,33 +91,75 @@ bool replace_symbolt::replace(exprt &dest) const
   if(!have_to_replace(dest))
     return result;
 
-  if(dest.id()==ID_symbol)
+  if(dest.id()==ID_member)
+  {
+    member_exprt &me=to_member_expr(dest);
+
+    if(!replace(me.struct_op(), replace_with_const))
+      result=false;
+  }
+  else if(dest.id()==ID_index)
+  {
+    index_exprt &ie=to_index_expr(dest);
+
+    if(!replace(ie.array(), replace_with_const))
+      result=false;
+
+    if(!replace(ie.index()))
```

Comment:
I'm not quite sure what you mean but it seems (from the diff and context -- please confirm @danpoe) that the replace_with_const flag is to ensure that an l-value is still a valid l-value after substitution.  For example if your expression is "a[i]" then substituting i for a constant results in an l-value but substituting a for a constant (array) wouldn't.